.golangci.yml

```diff
@@ -1,27 +1,112 @@
-run:
-  timeout: 2m
+linters-settings:
+  funlen:
+    lines: 100
+    statements: 50
+  goconst:
+    min-len: 2
+    min-occurrences: 2
+  govet:
+    check-shadowing: true
+    settings:
+      printf:
+        funcs:
+          - (github.com/golangci/golangci-lint/pkg/logutils.Log).Infof
+          - (github.com/golangci/golangci-lint/pkg/logutils.Log).Warnf
+          - (github.com/golangci/golangci-lint/pkg/logutils.Log).Errorf
+          - (github.com/golangci/golangci-lint/pkg/logutils.Log).Fatalf
+  golint:
+    min-confidence: 0
+  gocyclo:
+    min-complexity: 8
+  maligned:
+    suggest-new: true
+  dupl:
+    threshold: 100
+  goconst:
+    min-len: 2
+    min-occurrences: 3
+  misspell:
+    locale: US
+  lll:
+    line-length: 160
+  goimports:
+    local-prefixes: github.com/gojekfarm/albatross
+  gocritic:
+    enabled-tags:
+      - performance
+      - style
+      - experimental
+    disabled-checks:
+      - wrapperFunc
+      - hugeParam
+      - octalLiteral
+
+    settings: # settings passed to gocritic
+      captLocal: # must be valid enabled check name
+        paramsOnly: true
+      rangeValCopy:
+        sizeThreshold: 200
 
 linters:
   disable-all: true
```

Comment:
This you can invert saying `enable-all: true` and disable specific plugins. if this's complete list, it's fine.

Reply:
The enable-all option has been deprecated, https://golangci-lint.run/usage/configuration/.
I have used the golang[-ci lint repo configuration](https://golangci-lint.run/usage/configuration/) as a reference that uses strict settings. I have included all linters they use, and also included some linters they have ommited (godot, gocognit, nestif for example). The linters I have not included are linters like (asciicheck and testpackage).